src/hashtable.py

```diff
@@ -51,8 +52,19 @@ def insert(self, key, value):
 
         Fill this in.
         '''
-        pass
 
+        hashed_key = self._hash_mod(key)
+        node = self.storage[hashed_key]
+
+        if node is None or node.key == key:
+            self.storage[hashed_key] = LinkedPair(key, value)
+            return
```

Comment:
I'd say there are some flaws in here that need to be addressed.  On line 60, you're placing a whole new node as the head, if the key matches.  If there were any nodes trailing the original node, they'll all be cut off.

The while loop that starts at 63 is travelling to the end of the list, but it's not checking the keys on the way, so if you have a match, nothing will be overwritten, you'll just be adding a new node at the end, which will never be able to retrieved, as there is a node w/ a matching key earlier in the list still.